app/src/main/java/app/revenge/manager/installer/step/installing/InstallStep.kt

```diff
@@ -38,7 +40,15 @@ class InstallStep(
             ?.takeIf { it.isNotEmpty() }
             ?: throw Error("Missing APKs from LSPatch step; failure likely")
 
-        val installer: Installer = when (preferences.installMethod) {
+        val installMethod = if (preferences.installMethod == InstallMethod.SHIZUKU && !ShizukuPermissions.waitShizukuPermissions()) {
+            // Temporarily use DEFAULT if SHIZUKU permissions are not granted
+            InstallMethod.DEFAULT
+        } else {
+            context.showToast(R.string.msg_shizuku_denied)
+            preferences.installMethod
```

Comment:
If perms are denied and the install method is SHIZUKU what happens?

Reply:
It uses preferences.installMethod which is the Android Package Installer

Reply:
Oh, I swapped them lol, oops